functions/index.js

```diff
@@ -1,9 +1,113 @@
 const functions = require("firebase-functions");
+const csv = require("csv-parser");
+const admin = require("firebase-admin");
+const moment = require("moment");
 
-// // Create and Deploy Your First Cloud Functions
-// // https://firebase.google.com/docs/functions/write-firebase-functions
-//
-// exports.helloWorld = functions.https.onRequest((request, response) => {
-//   functions.logger.info("Hello logs!", {structuredData: true});
-//   response.send("Hello from Firebase!");
-// });
+admin.initializeApp();
+
+const solarArraysRef = admin.firestore().collection("Solar Arrays");
+
+exports.onFileUpload = functions.storage.object().onFinalize(async (object) => {
+  try {
+    const fileBucket = object.bucket;
+    const filePath = object.name;
+
+    if (!filePath.endsWith(".csv")) {
+      console.log(`File ${filePath} is not in CSV format, skipping...`);
+      return false;
+    }
+
+    // create a reference to the Cloud Storage file and set up a CSV parser
+    const bucket = admin.storage().bucket(fileBucket);
+    const file = bucket.file(filePath);
+    const csvStream = csv();
+
+    // Determine the collection name based on the file path.
+    const collectionName =
+      filePath.toLowerCase().includes("gordon") ||
+      filePath.toLowerCase().includes("arboretum")
```

Comment:
Potential arrays should not be hard coded (in case more arrays are added in the future). You should query the whole Solar Arrays collection check whether the name of the file matches the "name" field in each document.